OpenRA.Mods.Common/Traits/World/PathFinder.cs

```diff
@@ -71,6 +72,8 @@ public PathFinder(World world)
 			if (sourcesList.Count == 1)
 			{
 				var source = sourcesList[0];
+				if (!world.Map.Contains(source))
```

Comment:
Do we need to check the sourcesList.Count > 1 case?

Reply:
No, `ToTargetCell` already guards with `Map.Contains` and an empty path is returned if the search fails.

Reply:
The `sourcesList.Count == 1` case also uses `ToTargetCell`, so why is this check necessary if its not below?

I feel that I may be missing something obvious here!

Reply:
Correct. However, we do need it for the `// For adjacent cells on the same layer, ...` case.

If the following scenario occurs, we need the guard to correctly return an empty path, rather than returning a path between the points.
- Source is outside the map.
- Target is inside the map.
- Source and target are adjacent (straddling the map edge)

I have moved the safety check within the condition for this, which should reduce the confusion.